daily_read/order_portal.py

```diff
@@ -76,20 +89,53 @@ def process_orders(self, closed_before_in_days=30):
             if order["identifier"] in self.projects_data.data.keys():
                 proj_info = self.projects_data.data[order["identifier"]]
                 if order["reports"]:
-                    proj_prog_rep = next(item for item in order["reports"] if item["name"] == "Project Progress")
-                    proj_info.report_iuid = proj_prog_rep["iuid"]
-                if proj_info.orderer in order_updates:
-                    order_updates[proj_info.orderer]["projects"].append(proj_info)
-                else:
+                    prog_reports = [item for item in order["reports"] if item["name"] == "Project Progress"]
+                    if len(prog_reports) > 1:
+                        log.error(f"Multiple reports for Project Progress for order {order['identifier']}")
+                    else:
+                        proj_info.report_iuid = prog_reports[0]["iuid"]
+
+                if proj_info.orderer not in order_updates:
                     order_updates[proj_info.orderer] = {
                         "pull_date": f"{pull_date}",
-                        "projects": [proj_info],
+                        "active_projects": 0,
+                        "recents": {},
+                        "projects": {},
                     }
 
+                latest_date = "0000-00-00"
+                latest_status = "None"
+                for date_value in proj_info.data["proj_dates"]:
+                    date_statuses = proj_info.data["proj_dates"][date_value]
```

Comment:
This should be the same right?

```suggestion
                for date_value, date_statuses in proj_info.data["proj_dates"].items():
```

Reply:
Done!